### DISCUSSION TITLE
Option to disable "New Formulae" and "New Casks" sections

### DISCUSSION THREAD

**Discussion by tyrann0us** [Homebrew/discussions#4590] [Everyday usage] [answered]
### Output of `brew config`

```shell
➜  ~ brew config
HOMEBREW_VERSION: 4.0.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54c8876dc39047c04de15e7d212979ae8d98cf1c
Last commit: 3 days ago
Core tap JSON: 22 Jun 05:21 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
➜  ~ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  php@7.4

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  php
```

### Description of issue

I've been using Homebrew for a couple of years now. My use case is that whenever I need a new command line tool or app, I google if it's available as a Homebrew formula or cask. On the result page (e.g., https://formulae.brew.sh/formula/composer), I then usually follow the link to the code on GitHub or check the stats, then copy & paste the install command.

Consequently, I never search using Homebrew directly, nor am I interested in the "New Formulae" and "New Casks" sections that show up when running `brew update`. For me, this information is irrelevant; I never found myself wanting to install a formula or cask just because it appeared in one of the sections.

But especially when not having run `brew update` for some time, these sections (and the "Deleted Formulae"/"Deleted Casks" counterparts) grow considerably, making it "hard" to find the "Outdated Formulae"/"Outdated Casks" sections, which for me is the only reason to run this command.

Long story short: I checked https://docs.brew.sh/Manpage#environment but could not find a way to disable the output of the mentioned sections. Is this possible differently? If not, it would be great to have an environment variable to control the output of `brew update`.

This is basically the counterpart of this discussion: https://github.com/orgs/Homebrew/discussions/4337.

Thanks!

**Comment by eugenesvk** [comment]
added 2 env vars that would allow suppressing this info noise 
`HOMEBREW_NO_UPDATE_REPORT_FORMULAE` and `HOMEBREW_NO_UPDATE_REPORT_CASKS`

https://github.com/Homebrew/brew/pull/15836

**Comment by MikeMcQuaid** [comment]
> which for me is the only reason to run this command.

Which command, `brew update`? It is run automatically for you so you shouldn't need to run it manually. If you only run it for outdated output: `brew outdated` seems more appropriate to run instead.

**Reply by eugenesvk** [reply]
I bumped into this issue when trying to make an update a bit faster to get a newly released version of a formula

**Reply by MikeMcQuaid** [reply]
@eugenesvk Do you have any before/after benchmarks with `hyperfine`? Given the implementation, I would be very surprised if this has a noticeable speed difference but would love to be proved wrong.

**Reply by eugenesvk** [reply]
I meant to manually force fetching the newest update, not that the update itself was faster

**Reply by osalbahr** [reply]
Is explicitly running `brew update` needed to install the latest formulae, even with install from API (the default)?

**Reply by MikeMcQuaid** [reply]
@eugenesvk gotcha, I see. For that case, FWIW, `brew update --auto` will hide the stuff you don't care about, too.

> Is explicitly running `brew update` needed to install the latest formulae, even with install from API (the default)?

If you want to override the auto-update check timers and really force an update: yes.

**Comment by danielbraun89** [comment, marked as answer]
The correct env is `HOMEBREW_NO_UPDATE_REPORT_NEW`
